interactive/dashboard/flow_diagram.py

```diff
@@ -119,153 +102,794 @@ def name_included_reactions(included_reactions, reactions_json):
+            total_quantites, reaction_names_on_hover)
+
+
+# create line widths scaling based on user selected option
+def calculateLineWeights(maxWidth, species_yields, scale_type):
+    # use fluxes from map we setup, scale the values, add weights and whatnot
+    maxVal = -1
+    minVal = 99999999
+    rawVal = {}
+    numOfInvalids = 0
+    if scale_type == 'log':
+        print("[5/7] logarithmically scaling weights...")
+        if species_yields != {}:
+
+            li = species_yields
+            logged = []
+            for i in li:
+                # fail safe for null->null value
+                if (str(i) != 'null->null'
+                        and float(0.0) != float(species_yields[i])):
+                    rawVal.update({i: species_yields[i]})
+                    try:
+                        logged.append((i, math.log(species_yields[i])))
+                        if species_yields[i] < minVal:
+                            minVal = species_yields[i]
+                        if species_yields[i] > maxVal:
+                            maxVal = species_yields[i]
+                    except ValueError:
+                        numOfInvalids = numOfInvalids+1
+                        logged.append(
+                            (i, abs(math.log(abs(species_yields[i])))))
+                        if abs(species_yields[i]) < minVal:
+                            minVal = abs(species_yields[i])
+                        if abs(species_yields[i]) > maxVal:
+                            maxVal = abs(species_yields[i])
+
+            vals = [i[1] for i in logged]
+            min_val = abs(min(vals))
+            range = max(vals) - min(vals)
+            if max(vals) == min(vals):
+                range = 1
+            scaled = [(x[0], (float(((x[1] + min_val)/range))
+                       * float(maxWidth)) + 1) for x in logged]
+            print("|_ found", numOfInvalids,
+                  "invalid values, reversed the lines for them")
+            print("[6/7] got logarithmically scaled weights")
+            return list(scaled), minVal, maxVal, rawVal
+        else:
+            print("[7/7] no species selected, returning empty list")
+            return list([]), -1, 999999999, {}
+    else:
+        print("[5/7] linearly scaling weights...")
+        li = species_yields
+        vals = [species_yields[i] for i in li]
+        min_val = abs(min(vals))
+
+        minVal = min_val
+        maxVal = abs(max(vals))
+
+        range = max(vals) - min(vals)
+        scaled = [(x, (((species_yields[x] + min_val)/range)
+                   * int(maxWidth)) + 1) for x in li]
+        print("[6/7] got linearly scaled weights")
+        return list(scaled), minVal, maxVal, {}
+
+
+# bring everything together, call functions and debug
+def new_find_reactions_and_species(list_of_species, reactions_data,
+                                   blockedSpecies, df, start,
+                                   end, max_width, scale_type):
```

Comment:
the description of this function could be clearer - maybe something like assemble set of edges and nodes with formatting options? is this correct? is this function doing anything else?